qds_sdk/cluster.py

```diff
@@ -1168,6 +1214,217 @@ def minimal_payload(self):
+        self.cloud_config['storage_config']['disk_storage_account_resource_group_name'] \
+            = disk_storage_account_resource_group_name
+
+    def set_engine_config(self, flavour=None, custom_hadoop_config =None, fairscheduler_config_xml=None,
+                          default_pool=None, use_qubole_placement_policy=None, presto_version=None,
+                          custom_presto_config=None, spark_version=None,
+                          custom_spark_config=None, dbtap_id=None, fernet_key=None, overrides=None):
+        self.engine_config['flavour'] = flavour
+        self.set_hadoop_settings(custom_hadoop_config, fairscheduler_config_xml, default_pool, use_qubole_placement_policy)
+        self.set_presto_settings(presto_version, custom_presto_config)
+        self.set_spark_settings(spark_version, custom_spark_config)
+        self.set_airflow_settings(dbtap_id, fernet_key, overrides)
+
+    def set_hadoop_settings(self, custom_hadoop_config =None, fairscheduler_config_xml=None, default_pool=None,
+                            use_qubole_placement_policy=None):
+        self.engine_config['hadoop_settings'] = {}
+        self.engine_config['hadoop_settings']['custom_hadoop_config'] = custom_hadoop_config
+        self.set_fair_scheduler_settings(fairscheduler_config_xml, default_pool)
+        self.engine_config['hadoop_settings']['use_qubole_placement_policy'] = use_qubole_placement_policy
+
+    def set_fair_scheduler_settings(self, fairscheduler_config_xml=None, default_pool=None):
+        self.engine_config['hadoop_settings']['fair_scheduler_settings'] = {}
+        self.engine_config['hadoop_settings']['fair_scheduler_settings']['fairscheduler_config_xml'] = fairscheduler_config_xml
+        self.engine_config['hadoop_settings']['fair_scheduler_settings']['default_pool'] = default_pool
+
+    def set_presto_settings(self, presto_version=None, custom_presto_config=None):
+        self.engine_config['presto_settings'] = {}
+        self.engine_config['presto_settings']['presto_version'] = presto_version
+        self.engine_config['presto_settings']['custom_presto_config'] = custom_presto_config
+
+    def set_spark_settings(self, spark_version=None, custom_spark_config=None):
+        self.engine_config['spark_settings'] = {}
+        self.engine_config['spark_settings']['spark_version'] = spark_version
+        self.engine_config['spark_settings']['custom_spark_config'] = custom_spark_config
+
+    def set_airflow_settings(self, dbtap_id=None, fernet_key=None, overrides=None):
+        self.engine_config['airflow_settings'] = {}
+        self.engine_config['airflow_settings']['dbtap_id'] = dbtap_id
+        self.engine_config['airflow_settings']['fernet_key'] = fernet_key
+        self.engine_config['airflow_settings']['overrides'] = overrides
+
+    def set_monitoring(self, ganglia=None, datadog_api_token=None, datadog_app_token=None):
```

Comment:
If the ganglia option is a boolean, let's call it `enable_ganglia`. Conveys what argument should be passed correctly.